File: Valid-Perfect-Square/VPS.py
class Solution:
    def isPerfectSquare(self, num):
        """
        :type num: int
        :rtype: bool
        """
        d = 2
        init = 1
        total = 0

        while total < num:
            print(total)
            total += init
            init += d

        if total == num:
            return True

        return False


a = Solution()

File: Valid-Perfect-Square/VPS.rs
fn isPerfectSquare(num: &i32) -> bool {
    let d = 2;
    let mut init = 1;
    let mut total = 0;

    while (total < *num) {
        total += init;
        init += d;
    }

    if total == *num {
        return true;
    }

    return false;
}

fn main() {
    println!("{:?}", isPerfectSquare(&14));
    println!("{:?}", isPerfectSquare(&16))
}
